src/rgw/rgw_rest.cc

```diff
@@ -1944,6 +1944,7 @@ int RGWHandler_REST::read_permissions(RGWOp* op_obj)
     if (!s->info.args.exists("tagging")){
       only_bucket = true;
     }
+    only_bucket = false;
```

Comment:
I think this should be fixed at L1919:  `bool only_bucket;`, so that there are no missing branches.